pkg/policymutation/policymutation.go

```diff
@@ -452,6 +470,10 @@ func CanAutoGen(policy *kyverno.ClusterPolicy, log logr.Logger) (applyAutoGen bo
 		}
 	}
 
+	if !needAutogen {
+		return false, "none"
```

Comment:
Returning "none" here is odd. If the first return is false, callers should ignore the second value, and you should return the zero value—an empty string—instead.

Reply:
We're using "none" in case `CanAutogen()` return false [here](https://github.com/kyverno/kyverno/blob/9e10eef42203b92d5d3c94a30a378d89c46ab29b/pkg/policymutation/policymutation.go#L354-L369). I think we can change this. I only see this one place using the second argument in this case.

Reply:
Yes, you should _not_ use that "desiredControllers" value if "applyAutoGen" is false. Instead, if "applyAutoGen" is false and you still want to set the "actualControllers" variable, do it conditionally.